src/utils.ts

```diff
@@ -20,9 +45,8 @@ export async function fetcher({
       body,
     }
     const response = await fetch(`${LICHESS_API_URL}${endpoint}`, fetchOptions)
-    const data = await response.json()
 
-    return data
+    return response
```

Comment:
The fetcher can take a parameter such as `header`, and determine whether to return as json or ndjson. By doing this, we can keep the current method: `return this.fetcher(...)`

Reply:
> The fetcher can take a parameter such as `header`, and determine whether to return as json or ndjson. By doing this, we can keep the current method: `return this.fetcher(...)`

I do use the fetcher to return the stream/ndjson. The problem is, the stream data is encoded and needs to be decoded. If we do not process the stream (decode the chunks, parse them into JSON) before returning from the function, the end programmer will have to decode the stream and parse it into JSON. 

With the `JSONStream`, the end programmer can do this...
```js
const following = await lichess.account.following // Return following users as a stream
for await (const user of following) {
    console.log(`You are following ${user.username}`)
    // Do stuff with user
}
```
instead of having to do this:
```js
const following = await lichess.account.following // Return following a stream
for await (const chunk of following) {
    const decoded = String(chunk)
    const user = JSON.parse(decoded)
    console.log(`You are following ${user.username}`)
    // Do stuff with user
} 
```

Personally, I prefer the first option, but it is your call.

Reply:
I meant why can't the fetcher handle data processing? Is there a reason why send the response to each method and let each handle one by one?

```ts
public async follow({ username }: { username: string }) {
  const response = await this.fetcher(`/api/rel/follow/${username}`, 'post')
  const json = await response.json() // DOING SAME JOB
  return json
}

public async unfollow({ username }: { username: string }) {
  const response = await this.fetcher(`/api/rel/unfollow/${username}`, 'post')
  const json = await response.json() // DOING SAME JOB
  return json
}
```

I am thinking something like:

```ts
this.fetcher({
  method: 'GET',
  url: `/api/study/by/${username}`,
  accept: 'application/x-ndjson', // let fetcher handle the incoming based on accept
})
```

and handle the streaming part inside the fetcher.

Also, since Lichess uses ndjson a lot, we could use the package like [`ndjson`](https://www.npmjs.com/package/ndjson).